main.go

```diff
@@ -4,20 +4,40 @@ import (
 	"WebCrawler/crawler"
 	"flag"
 	"log"
+	"net/url"
 	"sync"
 )
 
 func main() {
-	url := flag.String("url", "", "url to start")
+	rawurl := flag.String("url", "", "url to start")
+	bound := flag.Bool("bound", false, "Domain to bound the crawler")
 
 	flag.Parse()
 
+	var filter crawler.Filter
+
+	url, err := url.Parse(*rawurl)
+
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	log.Println(*bound)
+
+	if *bound {
+		log.Println(url.Hostname())
+		filter = crawler.CrossDomainFilter{url.Hostname()}
+	} else {
+		filter = crawler.NoneFilter{}
```

Comment:
Does go have DI inbuilt or any framework?
😆

Reply:
We can add a static DI frameworks but they are not needed for the filters and they are only made once.